app/src/main/java/xlang/parser/statement/NullStatement.java

```diff
@@ -0,0 +1,9 @@
+package xlang.parser.statement;
+
+public class NullStatement implements Statement {
+
+    @Override
+    public void execute() {
```

Comment:
it should return null here. because we cannot evaluate void and also cannot pass it to other methods

Reply:
This is null Statement not Expression. It is used as null object pattern when `else` block is not passed to `IfElse` statement. 
I am thinking to not add null expression value in this language.

Reply:
Then how will you handle undefined values in the language?

Reply:
Good question.
why null/undefiend/None is bad  : https://www.lucidchart.com/techblog/2015/08/31/the-worst-mistake-of-computer-science/

But in real world we have to handle cases where the value is not present, such as in database;

Reply:
I think what is wrong with null, allowing all the value to be implicitly null. 
for example in java when we define a String variable
```
 String name;
```
It is a lie, name variable can have String value and special case null value.
So a variable has always two faces on is it's type and other is null type.

Reply:
The best way to handle `null` is, make it explict to programmer to say this value can be null. 
Most programming language use optional type pattern to handle this. 

Here's an incomplete list of languages that are null-safe in the sense that they don't have any non-nonnullable types:

 [Dart](https://dart.dev/null-safety) (2021): Has optional types with ? syntax.
    [C# 8](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8) (2019): Has opt-in "nullable reference types".
    [Kotlin](https://kotlinlang.org/) (2015): Has optional types with ? syntax.
    [Pony](http://www.ponylang.org/) (2015). Uses union type where one of the types is None.
    [Swift](https://swift.org/) (2014): Has optional types with ? syntax.
    [Crystal](http://crystal-lang.org/) (2014): Does have nil, but prevents all null pointer exceptions at compile-time.
    [Hack](http://hacklang.org/) (2014): Has optional types with ? syntax.
    [TypeScript](https://www.typescriptlang.org/) (2012): Has union types that can have undefined or null as a variant.
    [Elm](http://elm-lang.org/) (2012): Has union type Maybe.
    [Ceylon](http://ceylon-lang.org/) (2011): Has optional types with ? syntax.
    [Rust](https://www.rust-lang.org/) (2010): Has optional type Option.
    [Fantom](http://fantom.org/) (2005): Has optional types with ? syntax.
    [F#](http://fsharp.org/) (2005): Has union type Option.
    [Nice](http://nice.sourceforge.net/) (2003): Has optional types with ? syntax.
    [Netlogo](https://groups.google.com/forum/#!topic/netlogo-devel/-v-dSIpFszA) (1999) has no type null
    [OCaml](https://ocaml.org/) (1996): Has union type option.
    [Haskell](https://www.haskell.org/) (1990): Has union type Maybe.
    [Standard ML](http://www.smlnj.org/sml.html) (1990): Has union type option.
    [Tcl](https://www.tcl.tk/) (1988)
    [Erlang](https://www.erlang.org/) (1986)
    [Prolog](https://en.wikipedia.org/wiki/Prolog) (1972): A logical variable stands for "anything at all". There is no concept of "null" or "undefined".